libgnucash/engine/Account.cpp

```diff
@@ -333,7 +334,7 @@ gnc_account_init(Account* acc)
     priv->lower_balance_limit = {};
     priv->include_sub_account_balances = {};
 
-    priv->splits = nullptr;
+    priv->splits = {};
```

Comment:
> `priv->splits.clear()` is better than assigning to an empty vector, but no initialization is needed, the `std::vector` constructor has already done that.

But the `AccountPrivate` object is created via `g_object_new` or `g_malloc` rather than `new` so the constructors won't get called?

Reply:
It all gets initialized to 0s, and that's what an empty vector is:
```
   frame #0: 0x00007ffff7ae6fd5 libgnc-engine.so`::gnc_account_init(acc=0x000055555561e1f0) at Account.cpp:303:20
   300 	    AccountPrivate *priv;
   301 	
   302 	    priv = GET_PRIVATE(acc);
-> 303 	    priv->parent   = nullptr;
   304 	    priv->children = nullptr;
   305 	
   306 	    priv->accountName = qof_string_cache_insert("");
(lldb) p sizeof(AccountPrivate)
(unsigned long) $1 = 360
(lldb) x/36wx priv
0x55555561e030: 0x00000000 0x00000000 0x00000000 0x00000000
0x55555561e040: 0x00000000 0x00000000 0x00000000 0x00000000
0x55555561e050: 0x00000000 0x00000000 0x00000000 0x00000000
0x55555561e060: 0x00000000 0x00000000 0x00000000 0x00000000
0x55555561e070: 0x00000000 0x00000000 0x00000000 0x00000000
0x55555561e080: 0x00000000 0x00000000 0x00000000 0x00000000
0x55555561e090: 0x00000000 0x00000000 0x00000000 0x00000000
0x55555561e0a0: 0x00000000 0x00000000 0x00000000 0x00000000
0x55555561e0b0: 0x00000000 0x00000000 0x00000000 0x00000000
...
   frame #0: 0x00007ffff7ae719a libgnc-engine.so`::gnc_account_init(acc=0x000055555561e1f0) at Account.cpp:333:40
   330 	
   331 	    priv->higher_balance_limit = {};
   332 	    priv->lower_balance_limit = {};
-> 333 	    priv->include_sub_account_balances = {};
   334 	
   335 	    priv->splits.clear();
   336 	    priv->sort_dirty = FALSE;
(lldb) p priv->splits
(std::vector<split_s *, std::allocator<> >) $5 = size=1 {
  std::_Vector_base<split_s *, std::allocator<> > = {
    _M_impl = {
      std::_Vector_base<split_s *, std::allocator<> >::_Vector_impl_data = {
        _M_start = nullptr
        _M_finish = nullptr
        _M_end_of_storage = nullptr
      }
    }
  }
}
...
Process 837193 stopped
* thread #1, name = 'test-engine', stop reason = step over
    frame #0: 0x00007ffff7ae71c7 libgnc-engine.so`::gnc_account_init(acc=0x000055555561e1f0) at Account.cpp:336:22
   333 	    priv->include_sub_account_balances = {};
   334 	
   335 	    priv->splits.clear();
-> 336 	    priv->sort_dirty = FALSE;
   337 	}
   338 	
   339 	static void
(lldb) p priv->splits
(std::vector<split_s *, std::allocator<> >) $6 = size=1 {
  std::_Vector_base<split_s *, std::allocator<> > = {
    _M_impl = {
      std::_Vector_base<split_s *, std::allocator<> >::_Vector_impl_data = {
        _M_start = nullptr
        _M_finish = nullptr
        _M_end_of_storage = nullptr
      }
    }
  }
}
```
So all of that setting stuff to `{}` is redundant, as is `priv->splits.clear()`.